src/usermeta.ts

```diff
@@ -587,6 +690,112 @@ class UsermetaRouter {
     const client = this.routes[key] || this.ganomedeLocal;
     return client.get(params, key, cb);
   }
+
+  getTasksForBulk(params: UsermetaClientBulkOptions | string, keys: string[], buildTask: (clientObj, _keys: string[]) => any[]): any[] {
+    let clients = {};
+    //sorting keys as per the client from routes.
+    //grouping each clients with its own keys.
+    keys.forEach((key) => {
+      const client = this.routes[key] || this.ganomedeLocal;
+      const clientType = client.type;
+      if (!clients[clientType]) {
+        clients[clientType] = {};
+        clients[clientType].keys = [];
+        clients[clientType].client = client;
+      }
+      clients[clientType].keys.push(key);
+    });
+
+    let tasks: any[] = [];
+    //generate the tasks for parallel
+    Object.keys(clients).forEach((clientType: string) => {
+      const clientObj = clients[clientType];
+      const _keys: string[] = clientObj.keys;
+      const _client = clientObj.client;
+      tasks = tasks.concat(buildTask(_client, _keys));
+    });
+    return tasks;
+  }
+
+  getBulk(params: UsermetaClientBulkOptions | string, keys: string[], cb: UsermetaClientBulkCallback) {
+    //generate tasks for get.
+    if (keys.length === 0)
+      return cb(null, []);
+    const tasks: any[] = this.getTasksForBulk(params, keys, (_client, _keys) => {
+      let _tasks: any[] = [];
+      //if getBulk method exists on this client ..  then we call getbulk method => and passing the keys
+      //related.
+      if (_client.getBulk) {
+        _tasks.push(cb1 => _client.getBulk(params, _keys, cb1));
+      }
+      else {
+        //else we loop over the keys => and we will create tasks based for each key => client =>call for key.
+        _keys.forEach((key: string) => {
+          _tasks.push(cb1 => _client.get(params, key, (err: Error | null, reply?: string | null) => {
+            if (err) {
+              return cb1(err, reply);
+            }
+            cb1(err, { key, value: reply });
+          }));
+        });
+      }
+      return _tasks;
+    });
+
+    async.parallel(tasks, (err, data) => {
+      if (err) {
+        cb(err, null);
+      }
+      else {
+        //make the 2 levels array to 1 level => [[], [], []] => []
+        let oneLevelArray = (data as any[][])?.flat();
```

Comment:
Here too, I hope we can get rid of this obscure `any[][]`. Especially as is goes into a variable (almost) called `array`, hard to understand what is in it...